### ISSUE TITLE
winesteam "Steam needs to be online to update."

### ISSUE THREAD

**Issue by proalexa** [lutris/lutris#2257]
**Describe the bug**

When trying to install [Heroes & Generals](https://lutris.net/games/heroes-generals/) with winesteam, Steam starts updating and then says: "Steam needs to be online to update. Please confirm your network connection and try again."

**Expected behavior**

Steam needs to work properly.

**Current behavior**

Steam - Fatal Error: "Steam needs to be online to update. Please confirm your network connection and try again."

**Steps to reproduce**

- Install [Heroes & Generals](https://lutris.net/games/heroes-generals/)

**Lutris debugging output**
```
INFO     2019-08-04 13:44:30,190 [application.do_command_line:225]:Running Lutris 0.5.2.2
INFO     2019-08-04 13:44:30,191 [startup.check_driver:54]:Using NVIDIA drivers 430.40 for x86_64
INFO     2019-08-04 13:44:30,191 [startup.check_driver:58]:GPU: GeForce MX150
INFO     2019-08-04 13:44:30,191 [startup.check_driver:75]:GPU: 10DE:1D10 17AA:39B0 using nvidia drivers
INFO     2019-08-04 13:44:30,191 [startup.check_driver:75]:GPU: 8086:5917 17AA:39B0 using i915 drivers
INFO     2019-08-04 13:44:30,217 [startup.check_vulkan:129]:Vulkan is supported
INFO     2019-08-04 13:44:30,218 [dxvk.get_dxvk_versions:24]:Updating DXVK versions
DEBUG    2019-08-04 13:44:30,412 [store.get_missing_media:182]:Requesting missing icons from API for krunker
INFO     2019-08-04 13:44:30,412 [lutriswindow.toggle_connection:562]:Connected to lutris.net as GrbavaCigla
DEBUG    2019-08-04 13:44:30,414 [http.get:68]:GET https://lutris.net/api/runtime
DEBUG    2019-08-04 13:44:30,421 [http.get:68]:GET https://lutris.net/api/games
DEBUG    2019-08-04 13:44:30,421 [http.get:68]:GET https://lutris.net/api/games/library/GrbavaCigla
DEBUG    2019-08-04 13:44:30,673 [api.get_game_api_page:132]:Loaded 1 games from page 1
DEBUG    2019-08-04 13:44:33,323 [interpreter.fetch_script:54]:Fetching installer https://lutris.net/api/installers/heroes-generals
DEBUG    2019-08-04 13:44:33,326 [http.get:68]:GET https://lutris.net/api/installers/heroes-generals
DEBUG    2019-08-04 13:44:37,421 [interpreter.check_runner_install:440]:Required runners: [<lutris.runners.wine.wine object at 0x7fb762b57950>, Winesteam runner (None)]
INFO     2019-08-04 13:44:37,425 [runner.get_runner_version:257]:Getting runner information for wine
DEBUG    2019-08-04 13:44:37,426 [http.get:68]:GET https://lutris.net/api/runners/wine
DEBUG    2019-08-04 13:44:37,558 [interpreter.check_runner_install:456]:Default wine version is ge-protonified-nofshack-4.9
INFO     2019-08-04 13:44:37,561 [runner.get_runner_version:257]:Getting runner information for winesteam
DEBUG    2019-08-04 13:44:37,562 [http.get:68]:GET https://lutris.net/api/runners/winesteam
ERROR    2019-08-04 13:44:37,794 [interpreter.check_runner_install:462]:Failed to get default wine version (got {})
DEBUG    2019-08-04 13:44:37,807 [interpreter._iter_commands:564]:Installer command: install_steam_game
WARNING  2019-08-04 13:44:37,823 [winesteam.get_game_path_from_appid:322]:Data path for SteamApp 227940 not found.
DEBUG    2019-08-04 13:44:37,824 [interpreter.install_steam_game:798]:Installing steam game 227940
DEBUG    2019-08-04 13:44:37,844 [system.execute:38]:Executing /home/cigla/.local/share/lutris/runners/wine/ge-protonified-4.10-x86_64/bin/wine /home/cigla/.local/share/lutris/runners/winesteam/prefix64/drive_c/Program Files (x86)/Steam/Steam.exe -no-cef-sandbox -console steam://install/227940
DEBUG    2019-08-04 13:44:37,844 [system.execute:44]:DRI_PRIME=1 STEAM_RUNTIME=/home/cigla/.local/share/lutris/runtime/steam LD_LIBRARY_PATH=/home/cigla/.local/share/lutris/runners/wine/ge-protonified-4.10-x86_64/lib:/home/cigla/.local/share/lutris/runners/wine/ge-protonified-4.10-x86_64/lib64:/lib:/lib64:/usr/lib:/usr/lib64:/usr/lib32:/usr/lib64:/home/cigla/.local/share/lutris/runtime/Ubuntu-18.04-i686:/home/cigla/.local/share/lutris/runtime/steam/i386/lib/i386-linux-gnu:/home/cigla/.local/share/lutris/runtime/steam/i386/lib:/home/cigla/.local/share/lutris/runtime/steam/i386/usr/lib/i386-linux-gnu:/home/cigla/.local/share/lutris/runtime/steam/i386/usr/lib:/home/cigla/.local/share/lutris/runtime/Ubuntu-18.04-x86_64:/home/cigla/.local/share/lutris/runtime/steam/amd64/lib/x86_64-linux-gnu:/home/cigla/.local/share/lutris/runtime/steam/amd64/lib:/home/cigla/.local/share/lutris/runtime/steam/amd64/usr/lib/x86_64-linux-gnu:/home/cigla/.local/share/lutris/runtime/steam/amd64/usr/lib:$LD_LIBRARY_PATH ANDROID_HOME=/opt/android-sdk COLORTERM=truecolor DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus DESKTOP_SESSION=gnome DISPLAY=:1 FLUTTER_HOME=/opt/flutter GDMSESSION=gnome GDM_LANG=en_US.UTF-8 GJS_DEBUG_OUTPUT=stderr GJS_DEBUG_TOPICS=JS ERROR;JS LOG GNOME_TERMINAL_SCREEN=/org/gnome/Terminal/screen/2880fc70_7ec4_4a78_944a_c2d2c53f4cb5 GNOME_TERMINAL_SERVICE=:1.191 HOME=/home/cigla LANG=en_US.UTF-8 LOGNAME=cigla MAIL=/var/spool/mail/cigla MOZ_PLUGIN_PATH=/usr/lib/mozilla/plugins PATH=/usr/local/bin:/usr/local/sbin:/usr/bin:/opt/android-sdk/tools:/opt/android-sdk/tools/bin:/opt/flutter/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl PWD=/home/cigla SESSION_MANAGER=local/Rakija:@/tmp/.ICE-unix/906,unix/Rakija:/tmp/.ICE-unix/906 SHELL=/usr/bin/fish SHLVL=1 SSH_AUTH_SOCK=/run/user/1000/keyring/ssh TERM=xterm-256color USER=cigla USERNAME=cigla VTE_VERSION=5603 WINDOWPATH=2 XAUTHORITY=/run/user/1000/gdm/Xauthority XDG_CURRENT_DESKTOP=GNOME XDG_DATA_DIRS=/home/cigla/.local/share/flatpak/exports/share/:/var/lib/flatpak/exports/share/:/usr/local/share/:/usr/share/ XDG_MENU_PREFIX=gnome- XDG_RUNTIME_DIR=/run/user/1000 XDG_SEAT=seat0 XDG_SESSION_CLASS=user XDG_SESSION_DESKTOP=gnome XDG_SESSION_ID=2 XDG_SESSION_TYPE=x11 XDG_VTNR=2 WINEDEBUG=-all WINEARCH=win64 WINE=/home/cigla/.local/share/lutris/runners/wine/ge-protonified-4.10-x86_64/bin/wine WINEPREFIX=/home/cigla/.local/share/lutris/runners/winesteam/prefix64 WINEESYNC=0
DEBUG    2019-08-04 13:44:46,116 [interpreter.revert:732]:Install cancelled
DEBUG    2019-08-04 13:44:46,118 [system.execute:38]:Executing pgrep Steam.exe$
INFO     2019-08-04 13:44:46,137 [winesteam.force_shutdown:408]:Waiting for Steam to shutdown...
INFO     2019-08-04 13:44:46,138 [winesteam.shutdown:464]:Shutting down Steam
DEBUG    2019-08-04 13:44:46,143 [command.start:111]:Running /usr/bin/lutris-wrapper 0 0 /home/cigla/.local/share/lutris/runners/wine/ge-protonified-4.10-x86_64/bin/wine /home/cigla/.local/share/lutris/runners/winesteam/prefix64/drive_c/Program Files (x86)/Steam/Steam.exe -no-cef-sandbox -console -shutdown
DEBUG    2019-08-04 13:44:46,143 [command.start:113]:ENV: DRI_PRIME="1"
DEBUG    2019-08-04 13:44:46,143 [command.start:113]:ENV: STEAM_RUNTIME="/home/cigla/.local/share/lutris/runtime/steam"
DEBUG    2019-08-04 13:44:46,143 [command.start:113]:ENV: LD_LIBRARY_PATH="/home/cigla/.local/share/lutris/runners/wine/ge-protonified-4.10-x86_64/lib:/home/cigla/.local/share/lutris/runners/wine/ge-protonified-4.10-x86_64/lib64:/lib:/lib64:/usr/lib:/usr/lib64:/usr/lib32:/usr/lib64:/home/cigla/.local/share/lutris/runtime/Ubuntu-18.04-i686:/home/cigla/.local/share/lutris/runtime/steam/i386/lib/i386-linux-gnu:/home/cigla/.local/share/lutris/runtime/steam/i386/lib:/home/cigla/.local/share/lutris/runtime/steam/i386/usr/lib/i386-linux-gnu:/home/cigla/.local/share/lutris/runtime/steam/i386/usr/lib:/home/cigla/.local/share/lutris/runtime/Ubuntu-18.04-x86_64:/home/cigla/.local/share/lutris/runtime/steam/amd64/lib/x86_64-linux-gnu:/home/cigla/.local/share/lutris/runtime/steam/amd64/lib:/home/cigla/.local/share/lutris/runtime/steam/amd64/usr/lib/x86_64-linux-gnu:/home/cigla/.local/share/lutris/runtime/steam/amd64/usr/lib:$LD_LIBRARY_PATH"
DEBUG    2019-08-04 13:44:46,143 [command.start:113]:ENV: WINEDEBUG="-all"
DEBUG    2019-08-04 13:44:46,143 [command.start:113]:ENV: WINEARCH="win64"
DEBUG    2019-08-04 13:44:46,143 [command.start:113]:ENV: WINE="/home/cigla/.local/share/lutris/runners/wine/ge-protonified-4.10-x86_64/bin/wine"
DEBUG    2019-08-04 13:44:46,143 [command.start:113]:ENV: WINEPREFIX="/home/cigla/.local/share/lutris/runners/winesteam/prefix64"
DEBUG    2019-08-04 13:44:46,143 [command.start:113]:ENV: WINEESYNC="0"
DEBUG    2019-08-04 13:44:46,144 [command.start:113]:ENV: PYTHONPATH="/usr/bin:/usr/lib/python37.zip:/usr/lib/python3.7:/usr/lib/python3.7/lib-dynload:/home/cigla/.local/lib/python3.7/site-packages:/usr/lib/python3.7/site-packages"
DEBUG    2019-08-04 13:44:46,148 [signals.register_handler:26]:Registering <bound method MonitoredCommand.on_stop of <lutris.command.MonitoredCommand object at 0x7fb763d04350>> to 22290
DEBUG    2019-08-04 13:44:47,149 [system.execute:38]:Executing pgrep Steam.exe$
DEBUG    2019-08-04 13:44:48,185 [system.execute:38]:Executing pgrep Steam.exe$
DEBUG    2019-08-04 13:44:49,225 [system.execute:38]:Executing pgrep Steam.exe$
DEBUG    2019-08-04 13:44:50,241 [system.execute:38]:Executing pgrep Steam.exe$
DEBUG    2019-08-04 13:44:51,255 [system.execute:38]:Executing pgrep Steam.exe$
DEBUG    2019-08-04 13:44:52,277 [system.execute:38]:Executing pgrep Steam.exe$
DEBUG    2019-08-04 13:44:53,299 [system.execute:38]:Executing pgrep Steam.exe$
DEBUG    2019-08-04 13:44:54,335 [system.execute:38]:Executing pgrep Steam.exe$
DEBUG    2019-08-04 13:44:55,374 [system.execute:38]:Executing pgrep Steam.exe$
DEBUG    2019-08-04 13:44:56,389 [system.execute:38]:Executing pgrep Steam.exe$
INFO     2019-08-04 13:44:56,404 [winesteam.force_shutdown:411]:Forcing Steam shutdown
DEBUG    2019-08-04 13:44:56,404 [system.execute:38]:Executing pgrep Steam.exe$
INFO     2019-08-04 13:44:56,417 [system.kill_pid:122]:Killing PID 21392
DEBUG    2019-08-04 13:44:57,418 [system.execute:38]:Executing pgrep Steam.exe$
DEBUG    2019-08-04 13:44:57,440 [command.start:111]:Running /usr/bin/lutris-wrapper 0 0 /home/cigla/.local/share/lutris/runners/wine/ge-protonified-4.10-x86_64/bin/wine /home/cigla/.local/share/lutris/runners/winesteam/prefix64/drive_c/Program Files (x86)/Steam/Steam.exe -no-cef-sandbox -console steam://uninstall/227940
DEBUG    2019-08-04 13:44:57,440 [command.start:113]:ENV: DRI_PRIME="1"
DEBUG    2019-08-04 13:44:57,440 [command.start:113]:ENV: STEAM_RUNTIME="/home/cigla/.local/share/lutris/runtime/steam"
DEBUG    2019-08-04 13:44:57,440 [command.start:113]:ENV: LD_LIBRARY_PATH="/home/cigla/.local/share/lutris/runners/wine/ge-protonified-4.10-x86_64/lib:/home/cigla/.local/share/lutris/runners/wine/ge-protonified-4.10-x86_64/lib64:/lib:/lib64:/usr/lib:/usr/lib64:/usr/lib32:/usr/lib64:/home/cigla/.local/share/lutris/runtime/Ubuntu-18.04-i686:/home/cigla/.local/share/lutris/runtime/steam/i386/lib/i386-linux-gnu:/home/cigla/.local/share/lutris/runtime/steam/i386/lib:/home/cigla/.local/share/lutris/runtime/steam/i386/usr/lib/i386-linux-gnu:/home/cigla/.local/share/lutris/runtime/steam/i386/usr/lib:/home/cigla/.local/share/lutris/runtime/Ubuntu-18.04-x86_64:/home/cigla/.local/share/lutris/runtime/steam/amd64/lib/x86_64-linux-gnu:/home/cigla/.local/share/lutris/runtime/steam/amd64/lib:/home/cigla/.local/share/lutris/runtime/steam/amd64/usr/lib/x86_64-linux-gnu:/home/cigla/.local/share/lutris/runtime/steam/amd64/usr/lib:$LD_LIBRARY_PATH"
DEBUG    2019-08-04 13:44:57,440 [command.start:113]:ENV: WINEDEBUG="-all"
DEBUG    2019-08-04 13:44:57,441 [command.start:113]:ENV: WINEARCH="win64"
DEBUG    2019-08-04 13:44:57,441 [command.start:113]:ENV: WINE="/home/cigla/.local/share/lutris/runners/wine/ge-protonified-4.10-x86_64/bin/wine"
DEBUG    2019-08-04 13:44:57,441 [command.start:113]:ENV: WINEPREFIX="/home/cigla/.local/share/lutris/runners/winesteam/prefix64"
DEBUG    2019-08-04 13:44:57,441 [command.start:113]:ENV: WINEESYNC="0"
DEBUG    2019-08-04 13:44:57,441 [command.start:113]:ENV: PYTHONPATH="/usr/bin:/usr/lib/python37.zip:/usr/lib/python3.7:/usr/lib/python3.7/lib-dynload:/home/cigla/.local/lib/python3.7/site-packages:/usr/lib/python3.7/site-packages"
DEBUG    2019-08-04 13:44:57,445 [signals.register_handler:26]:Registering <bound method MonitoredCommand.on_stop of <lutris.command.MonitoredCommand object at 0x7fb763c328d0>> to 22313
DEBUG    2019-08-04 13:44:57,446 [system.remove_folder:197]:Removing folder /home/cigla/.cache/lutris/installer/heroes-generals
Running /home/cigla/.local/share/lutris/runners/wine/ge-protonified-4.10-x86_64/bin/wine /home/cigla/.local/share/lutris/runners/winesteam/prefix64/drive_c/Program Files (x86)/Steam/Steam.exe -no-cef-sandbox -console -shutdown
ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
ILocalize::AddFile() failed to load file "public/steambootstrapper_unknown.txt".
Aborting HTTP request because we received a response with a content-length over 64M and we are not streamingthreadtools.cpp (2910) : Assertion Failed: Illegal termination of worker thread 'Thread(0x00731C30/0x0000009C)'
Assert( Assertion Failed: Illegal termination of worker thread 'Thread(0x00731C30/0x0000009C)' ):threadtools.cpp:2910

Waiting on children
Exit with returncode 7
DEBUG    2019-08-04 13:44:57,449 [command.on_stop:158]:The process has terminated with code 1792
DEBUG    2019-08-04 13:44:57,449 [command.stop:241]:Management process looks dead already.
DEBUG    2019-08-04 13:44:57,449 [command.stop:249]:Detaching logger
Running /home/cigla/.local/share/lutris/runners/wine/ge-protonified-4.10-x86_64/bin/wine /home/cigla/.local/share/lutris/runners/winesteam/prefix64/drive_c/Program Files (x86)/Steam/Steam.exe -no-cef-sandbox -console steam://uninstall/227940
ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
ILocalize::AddFile() failed to load file "public/steambootstrapper_unknown.txt".
[2019-08-04 13:44:57] Startup
[2019-08-04 13:44:58] Verifying installation...
[2019-08-04 13:44:58] Downloading Update...
[2019-08-04 13:44:58] Checking for available update...
[2019-08-04 13:44:58] Package file tenfoot_fonts_all.zip.12f7f1fa582860ea27358593f7082b3db603e444 missing or incorrect size

[2019-08-04 13:44:58] Package file tenfoot_images_all.zip.f44393217929454515a70c08d6e2f30c45493c47 missing or incorrect size

[2019-08-04 13:44:58] Package file resources_all.zip.c49ec9913ac8c38561a57a19b4bb13b15d92844c missing or incorrect size

[2019-08-04 13:44:58] Package file bins_win32.zip.f31571fcdff23f3e30ebc658161fc2e9d74603f7 missing or incorrect size

[2019-08-04 13:44:58] Downloading update (0 of 170,344 KB)...
Aborting HTTP request because we received a response with a content-length over 64M and we are not streaming[2019-08-04 13:44:58] Error: Download of package (resources_all) failed after 0 bytes (200).

[2019-08-04 13:44:58] Download Complete.
[2019-08-04 13:44:58] Error: Steam needs to be online to update.  Please confirm your network connection and try again.

[2019-08-04 13:45:00] Shutdown
threadtools.cpp (2910) : Assertion Failed: Illegal termination of worker thread 'Thread(0x00731C40/0x000000A8)'
Assert( Assertion Failed: Illegal termination of worker thread 'Thread(0x00731C40/0x000000A8)' ):threadtools.cpp:2910

Waiting on children
All children gone
Exit with returncode 7
DEBUG    2019-08-04 13:45:02,526 [command.on_stop:158]:The process has terminated with code 1792
DEBUG    2019-08-04 13:45:02,526 [command.stop:241]:Management process looks dead already.
DEBUG    2019-08-04 13:45:02,526 [command.stop:253]:logger already detached
```

**System information**

[lutris-issue-report.zip](https://github.com/lutris/lutris/files/3464866/lutris-issue-report.zip)
Github cannot upload JSON so I uploaded zip.

**Screenshots**
![Screenshot from 2019-08-04 13-51-14](https://user-images.githubusercontent.com/23121814/62423227-f4e1f000-b6be-11e9-931e-4dd8471f0329.png)
![Screenshot from 2019-08-04 13-51-52](https://user-images.githubusercontent.com/23121814/62423235-07f4c000-b6bf-11e9-8b99-5c8d13d274e8.png)
![Screenshot from 2019-08-04 13-52-17](https://user-images.githubusercontent.com/23121814/62423247-180c9f80-b6bf-11e9-913f-dc83cab6116d.png)

**Comment by tannisroot** [comment]
Is this still an issue? IIRC Steam and internet in general weren't feeling so great recently, so that could have been the reason.
Also make sure you installed all the dependencies from this page: https://github.com/lutris/lutris/wiki/Wine-Dependencies

**Comment by tannisroot** [comment]
Duplicate of https://github.com/lutris/lutris/issues/2257
